rotkehlchen/chain/ethereum/adex/adex.py

```diff
@@ -0,0 +1,352 @@
+        https://github.com/python/mypy/issues/7781
+        """
+        tx_hash = HexStr(raw_event['id'])
+        identity_address = to_checksum_address(raw_event['owner'])
+        address = identity_address_map[identity_address]
+        bond_id = HexStr(raw_event['bondId'])
+        timestamp = Timestamp(raw_event['timestamp'])
+
+        if case == 'bond':
+            amount = FVal(raw_event['amount']) / ADX_AMOUNT_MANTISSA
+            return Bond(
+                tx_hash=tx_hash,
+                address=address,
+                identity_address=identity_address,
+                bond_id=bond_id,
+                amount=amount,
+                pool_id=HexStr(raw_event['poolId']),
+                slashed_at_start=Timestamp(int(raw_event['slashedAtStart'])),
+                timestamp=timestamp,
+            )
+        if case == 'unbond':
+            return Unbond(
+                tx_hash=tx_hash,
+                address=address,
+                identity_address=identity_address,
+                bond_id=bond_id,
+                timestamp=timestamp,
+            )
+        if case == 'unbond_request':
+            will_unlock = FVal(raw_event['willUnlock']) / ADX_AMOUNT_MANTISSA
+            return UnbondRequest(
+                tx_hash=tx_hash,
+                address=address,
+                identity_address=identity_address,
+                bond_id=bond_id,
+                will_unlock=will_unlock,
+                timestamp=timestamp,
+            )
+        raise AssertionError(f'Unexpected AdEx case: {case}.')
+
+    @overload  # noqa: F811
+    def _get_balances_graph(  # pylint: disable=no-self-use
+            self,
+            addresses: List[ChecksumEthAddress],
+            case: Literal['bonds'],
+    ) -> List[Bond]:
+        ...
+
+    @overload  # noqa: F811
+    def _get_balances_graph(  # pylint: disable=no-self-use
+            self,
+            addresses: List[ChecksumEthAddress],
+            case: Literal['unbonds'],
+    ) -> List[Unbond]:
+        ...
+
+    @overload  # noqa: F811
+    def _get_balances_graph(  # pylint: disable=no-self-use
+            self,
+            addresses: List[ChecksumEthAddress],
+            case: Literal['unbond_requests'],
+    ) -> List[UnbondRequest]:
+        ...
+
+    def _get_balances_graph(
+            self,
+            addresses: List[ChecksumEthAddress],
+            case: Literal['bonds', 'unbonds', 'unbond_requests'],
+    ) -> Union[List[Bond], List[Unbond], List[UnbondRequest]]:
+        """Get the addresses' events data querying the AdEx subgraph.
+        """
+        identity_address_map = (
+            self._get_identity_address_map(addresses)
+        )
+        # TODO: fix type for partial
+        # deserialization_method: Callable[[Any], Any]
+        querystr: str
+        schema: Literal['bonds', 'unbonds', 'unbondRequests']
+        if case == 'bonds':
+            deserialization_method = partial(
```

Comment:
If you don't like the use of `partial` I can change it... well, it is a nightmare for types.

I'd like to keep a single deserialization method as the events share many attributes.

Reply:
Please don't use partial here. I don't understand why you would do that. This is not a usecase for partial.

You need to have different deserialization functions so that typing is also correct and nice, and then each one can be calling the function that handles the bulk of the common work.

But please no don't do that with partial, it makes no sense here